sys/net/link_layer/ieee802154/ieee802154.c

```diff
@@ -46,46 +46,46 @@ size_t ieee802154_set_frame_hdr(uint8_t *buf, const uint8_t *src, size_t src_len
     /* set sequence number */
     buf[2] = seq;
 
-    if (bcast || (dst_len != 0)) {
+    if (dst_len != 0) {
         buf[pos++] = dst_pan.u8[0];
         buf[pos++] = dst_pan.u8[1];
     }
 
     /* fill in destination address */
-    if (bcast) {
-        /* no ACK_REQ for broadcast */
-        buf[0] &= ~IEEE802154_FCF_ACK_REQ;
-        buf[1] &= ~IEEE802154_FCF_DST_ADDR_MASK;
-        buf[1] |= IEEE802154_FCF_DST_ADDR_SHORT;
-        buf[pos++] = 0xff;
-        buf[pos++] = 0xff;
-    }
-    else {
-        switch (dst_len) {
-            case 0:
-                buf[1] |= IEEE802154_FCF_DST_ADDR_VOID;
-                break;
-            case 2:
-                buf[1] |= IEEE802154_FCF_DST_ADDR_SHORT;
-                buf[pos++] = dst[1];
-                buf[pos++] = dst[0];
-                break;
-            case 8:
-                buf[1] |= IEEE802154_FCF_DST_ADDR_LONG;
-                for (int i = 7; i >= 0; i--) {
-                    buf[pos++] = dst[i];
-                }
-                break;
-            default:
-                return 0;
-        }
+    switch (dst_len) {
+        case 0:
+            buf[1] |= IEEE802154_FCF_DST_ADDR_VOID;
+            break;
+        case 2:
+            if (memcmp(dst, ieee802154_addr_bcast,
```

Comment:
This should be `if(!memcmp(...)) ...`, currently this PR introduces ACKs for broadcasts, which is _unwise_.